dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/Governors/Standard/Generic/GovCT2.mo

```diff
@@ -0,0 +1,272 @@
+  Modelica.Blocks.Sources.RealExpression omegaPu4(y = omegaPu) annotation( Placement(visible = true, transformation(origin = {149, 227}, extent = {{-13, -9}, {13, 9}}, rotation = 0)));
+  Modelica.Blocks.Math.Product prodCfe annotation( Placement(visible = true, transformation(origin = {252, -54}, extent = {{8, -8}, {-8, 8}}, rotation = -90)));
+  Modelica.Blocks.Math.Product prodOmegaCfe annotation( Placement(visible = true, transformation(origin = {184, 114}, extent = {{-8, -8}, {8, 8}}, rotation = -180)));
+  Modelica.Blocks.Math.Product prodOmegaDm annotation( Placement(visible = true, transformation(origin = {236, 180}, extent = {{8, -8}, {-8, 8}}, rotation = 180)));
+  Modelica.Blocks.Nonlinear.SlewRateLimiter rateLimitFsrt(Falling = RDownPu,Rising = RUpPu, Td = tDRatelim, initType = Modelica.Blocks.Types.Init.InitialOutput, y_start = initFsrtPu) annotation( Placement(visible = true, transformation(origin = {-4, 40}, extent = {{-8, -8}, {8, 8}}, rotation = 0)));
+  Modelica.Blocks.Nonlinear.SlewRateLimiter rateLimitPRate(Rising = PRatePu, Td = tDRatelim, initType = Modelica.Blocks.Types.Init.InitialOutput, y_start = (ValveMaxPu - WFnlPu) * KTurb)  annotation( Placement(visible = true, transformation(origin = {138, 14}, extent = {{-8, -8}, {8, 8}}, rotation = -90)));
+  Modelica.Blocks.Logical.Switch switchWFSpd annotation( Placement(visible = true, transformation(origin = {272, -126}, extent = {{10, -10}, {-10, 10}}, rotation = -90)));
+  Modelica.Blocks.Tables.CombiTable1Ds tablePLimFromf(extrapolation = Modelica.Blocks.Types.Extrapolation.LastTwoPoints, smoothness = Modelica.Blocks.Types.Smoothness.LinearSegments, table = PLimFromfPoints, tableOnFile = false, verboseRead = false) annotation( Placement(visible = true, transformation(origin = {138, 38}, extent = {{-10, -10}, {10, 10}}, rotation = -90)));
+  Modelica.Blocks.Continuous.TransferFunction transferFunctCtB(a = {tB, 1}, b = {tC, 1}, initType = Modelica.Blocks.Types.Init.InitialState, x_start = {initPMechNoLossPu}, y_start = initPMechNoLossPu) annotation( Placement(visible = true, transformation(origin = {254, 106}, extent = {{10, -10}, {-10, 10}}, rotation = -90)));
+  Modelica.Blocks.Continuous.TransferFunction transferFunctSAtSB(a = {tSB, 1}, b = {tSA, 1}, initType = Modelica.Blocks.Types.Init.InitialOutput, y_start = initTexPu) annotation( Placement(visible = true, transformation(origin = {80, 114}, extent = {{10, -10}, {-10, 10}}, rotation = 0)));
+  // initial parameters
+  parameter Types.AngularVelocityPu omega0Pu = SystemBase.omega0Pu "Initial rotor speed";
+  final parameter Types.ActivePowerPu PRef0Pu = RDroop*(if RSelectInt == 0 then 0 elseif RSelectInt == 1 then Pm0Pu else initValvePu) "Initial value of reference power";
+  parameter Types.ActivePowerPu PGen0Pu = Pm0Pu*PBaseMw/PGenBaseMw "Initial value of generated active power in pu (base SnRef (system base), generator convention";
```

Comment:
the preassembled uses 
`initName="Dynawo.Electrical.Controls.Machines.Governors.GovernorPmPGen_INIT"`

This defines PGen0Pu as a signal. 

How do I fix this? Shall I just remove PGen0Pu from the INIT, so that the definition in GovCT2.mo is used instead?

Reply:
No just remove the definition in GovCT2.mo because the one coming from the generator should be used

Reply:
ah, now I got you